homeassistant/components/nws_alerts/sensor.py

```diff
@@ -0,0 +1,179 @@
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import API_ENDPOINT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the sensor platform."""
+
+    async def async_update_data():
+        """Fetch data from OWM."""
+        # Using a data update coordinator so we don't literally get rid of all our requests for the month >.>
+        endpoint = API_ENDPOINT.format(
+            lat=config_entry.data["lat"],
+            lon=config_entry.data["lon"],
+            api_key=config_entry.data["api_key"],
+        )
+        try:
+            response = await hass.async_add_executor_job(requests.get, endpoint)
+        except requests.exceptions.HTTPError as error:
+            raise UpdateFailed(
+                "Cannot connect to alerts API. Please try again later"
+            ) from error
+        except requests.exceptions.RequestException as error:
+            raise UpdateFailed(
+                "Cannot connect to alerts API. Please try again later"
+            ) from error
+
+        # check if it didn't return code 401
+        if response.status_code == 401:
+            _LOGGER.error("Invalid API key")
+            raise ConfigEntryAuthFailed("Invalid API key")
+
+        if response.status_code == 404:
+            _LOGGER.error("Invalid location")
+            raise ConfigEntryAuthFailed("Invalid location")
+
+        if response.status_code == 429:
+            _LOGGER.error("Too many requests")
+            raise UpdateFailed("Too many requests")
+
+        if (
+            response.status_code == 500
+            or response.status_code == 502
+            or response.status_code == 503
+            or response.status_code == 504
+        ):
+            _LOGGER.error("Service Unavailable")
+            raise UpdateFailed("Service Unavailable")
+
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise UpdateFailed("Unknown error")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="nws_alerts",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=config_entry.data.get("update_interval")),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+    sensor = WeatherAlertSensor(hass, config_entry, coordinator)
+    async_add_entities([sensor])
+
+
+class WeatherAlertSensor(CoordinatorEntity, SensorEntity):
+    """Weather alert sensor."""
+
+    def __init__(self, hass, config, coordinator):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.hass = hass
+        self._name = config.data.get("friendly_name", "NWS Alerts")
+        self._alert_count = None
+        self._unique_id = (
```

Comment:
Use attribute shorthand for name and unique_id:
```suggestion
        self._attr_name = config.data.get("friendly_name", "NWS Alerts")
        self._alert_count = None
        self._attr_unique_id = (
```